x-pack/plugins/gis/public/components/map/mb/view.js

```diff
@@ -185,9 +185,22 @@ export class MBMapContainer extends React.Component {
     if (!isMapReady) {
       return;
     }
+
     removeOrphanedSourcesAndLayers(this._mbMap, layerList);
-    layerList.forEach((layer) => {
-      layer.syncLayerWithMB(this._mbMap);
+    layerList.forEach(layer => {
+      if (!layer.dataHasLoadError()) {
+        Promise.resolve(layer.syncLayerWithMB(this._mbMap))
+          .catch(({ message }) => {
+            switch(layer._descriptor.type) {
```

Comment:
Is this switch needed? I would say all errors have the same action - update the layer state with the error message